addon/macro.js

```diff
@@ -3,29 +3,72 @@
  * https://github.com/jamesarosen/ember-i18n/blob/master/addon/utils/macro.js
  */
 import { assert } from '@ember/debug';
-import { computed, get } from '@ember/object';
+import { computed, get, getProperties } from '@ember/object';
 import { getOwner } from '@ember/application';
 import EmptyObject from 'ember-intl/-private/empty-object';
 
-const { keys } = Object;
+function partitionDynamicValuesAndStaticValues(options) {
+  const dynamicValues = new EmptyObject();
+  const staticValues = new EmptyObject();
 
-function values(object) {
-  return keys(object).map(key => object[key]);
+  Object.keys(options).forEach(key => {
+    const value = options[key];
+    if (value instanceof Raw) {
+      staticValues[key] = value;
```

Comment:
Could potentially already eagerly call `value.valueOf()` here to allow the `Raw` instance to be garbage collected and avoid repeated (but lazy) type coercion every time the macro recomputes.